src/ipyvizzustory/storylib/story.py

```diff
@@ -132,22 +132,65 @@ def add_step(self, step: Step) -> None:
 class StorySize:
     """A class for representing the size of a presentation story."""
 
-    def __init__(self, width: Optional[str] = None, height: Optional[str] = None):
+    def __init__(
+        self,
+        width: Optional[Union[int, float, str]] = None,
+        height: Optional[Union[int, float, str]] = None,
+        aspect_ratio: Optional[Union[int, float, str]] = None,
+    ):
         """
         StorySize constructor.
 
         Args:
             width: The width of a presentation story.
             height: The height of a presentation story.
+            aspect_ratio: The aspect-ratio of a presentation story.
+
+        Raises:
+            ValueError: If aspect_ratio and height are set together.
         """
+
+        width = self._convert_to_pixel_or_return(width)
+        height = self._convert_to_pixel_or_return(height)
+
         self._width = width
         self._height = height
+        self._aspect_ratio = aspect_ratio
 
         self._style = ""
-        if any([width is not None, height is not None]):
-            width = "" if width is None else f"width: {width};"
-            height = "" if height is None else f"height: {height};"
-            self._style = f"vp.style.cssText = '{width}{height}'"
+        if height is not None and aspect_ratio is not None:
+            raise ValueError("height and aspect ratio cannot be set at the same time")
+        if any([width is not None, height is not None, aspect_ratio is not None]):
+            _width = "" if width is None else f"width: {width};"
+            _height = "" if height is None else f"height: {height};"
+            _aspect_ratio = (
+                ""
+                if aspect_ratio is None
+                else f"aspect-ratio: {aspect_ratio} !important;"
+            )
+            self._style = f"vp.style.cssText = '{_aspect_ratio}{_width}{_height}'"
+
+    @staticmethod
+    def _convert_to_pixel_or_return(value: Any) -> Optional[str]:
+        if StorySize._is_int(value) or StorySize._is_float(value):
+            return str(value) + "px"
```

Comment:
With this, I can create a '-infpx' / 'nanpx' if I pass '-inf' / 'nan' as argument for width/height, because this string can be converted to float.

Reply:
fixed in [fe2a044](https://github.com/vizzuhq/ipyvizzu-story/pull/75/commits/fe2a044e6ce063d3407ad85c59bc121cb8e1a0d4)